runtime/onert/core/src/compiler/train/StaticBackPropShapeInferer.cc

```diff
@@ -60,7 +60,7 @@ void StaticBackPropShapeInferer::dump()
 bool StaticBackPropShapeInferer::checkDynamicInput(const ir::IOperation &op)
 {
   const auto &operands = _lowered_subg->graph().operands();
-  for (auto input_idx : op.getInputs() | ir::Remove::UNDEFINED | ir::Remove::DUPLICATED)
+  for (auto &input_idx : op.getInputs() | ir::Remove::UNDEFINED | ir::Remove::DUPLICATED)
```

Comment:
How about adding `const` like below? It will help prevent potential issue modifying `input_idx`.
```suggestion
  for (const auto &input_idx : op.getInputs() | ir::Remove::UNDEFINED | ir::Remove::DUPLICATED)
```

Reply:
Good. I'll update.